docs/advanced/pyproject_toml.md

```diff
@@ -103,6 +103,49 @@ matplotlib = "*"
 This would result in the conda dependencies being installed and the pypi dependencies being ignored.
 As pixi takes the conda dependencies over the pypi dependencies.
 
+## Optional dependencies
+If your python project includes groups of optional dependencies, pixi will automatically interpret them as [pixi features](../configuration.md#the-feature-table) of the same name with the associated `pypi-dependencies`.
+
+You can add them to pixi environments manually, or use `pixi init` to setup the project, which will create one environment per feature. Self-references to other groups of optional dependencies are also handled.
+
+For instance, imagine you have a project folder with a `pyproject.toml` file similar to:
+```toml
+[project]
+name = "my_project"
+dependencies = ["package1"]
+
+[project.optional-dependencies]
+test = ["pytest"]
+all = ["package2","my_project[test]"]
+```
+
+Running `pixi init` in that project folder will transform the `pyproject.toml` file into:
+```toml
+[project]
+name = "my_project"
+dependencies = ["package1"]
+
+[project.optional-dependencies]
+test = ["pytest"]
+all = ["package2","my_project[test]"]
+
+[tool.pixi.project]
+channels = ["conda-forge"]
+platforms = ["linux-64"] # if executed on linux
+
+[tool.pixi.environments]
+default = {features = [], solve-group = "default"}
+test = {features = ["test"], solve-group = "default"}
+all = {features = ["all", "test"], solve-group = "default"}
+```
+In this example, three environments will be created by pixi:
+
+ - **default** with 'package1' as pypi dependency
+ - **test** with 'package1' and 'pytest' as pypi dependencies
+ - **all** with 'package1', 'package2' and 'pytest' as pypi dependencies
```

Comment:
With the above, is it correct that installing the `test` environment with `pixi` would install `pytest` from `pypi` rather than `conda-forge`?

If I wanted my `pixi` `test` environment to only install dependencies from `conda-forge` can I specify that?

Reply:
@dhirschfeld you are correct.

Currently, you would have to manually add them to the `tool.pixi.feature.test.dependencies` table. What you want is also on my wishlist :-) https://github.com/prefix-dev/pixi/issues/532

Reply:
So, if I did manually specify the test dependencies in `[tool.pixi.feature.test.dependencies]` table, then that would be used ***instead*** of mapping `pypi` dependencies from the `[project.optional-dependencies]` table?

Reply:
Yes. Pixi currently solves pypi dependencies after the conda ones. And if it detects a package is already provided by one of the conda packages requested, it is skipped.